packages/functions/src/speaking.ts

```diff
@@ -3,119 +3,173 @@ import {
+        Media: {
+          MediaFileUri: `s3://${uploadResponseBucket}/${audioFileName}.webm`,
+        },
+        OutputBucketName: uploadResponseBucket,
       }),
     );
-    if (!commandOutput.Body) {
-      return;
-    }
-    const content = await commandOutput.Body.transformToString('utf-8');
-    if (content.length === 0) {
-      console.log(`file ${key} is empty`);
-    } else {
-      const answer = JSON.parse(content).results.transcripts[0].transcript;
-      const prompt = createPrompt(rubric, question, answer);
-      const input = {
-        inputText: prompt,
-        textGenerationConfig: {
-          maxTokenCount: 4096,
-          stopSequences: [],
-          temperature: 0,
-          topP: 0.9,
-        },
-      };
-
-      const command = new InvokeModelCommand({
-        body: JSON.stringify(input),
-        contentType: 'application/json',
-        accept: '*/*',
-        modelId: 'amazon.titan-text-express-v1',
-      });
-
-      const response = await bedrockClient.send(command);
-      const response_byte = response.body;
-
-      const textDecoder = new TextDecoder('utf-8');
-      const decodedString = textDecoder.decode(response_byte);
-
-      const feedbackResult = JSON.parse(decodedString).results[0].outputText;
-
-      const score_index = feedbackResult.indexOf('Score:');
-      const feedback_index = feedbackResult.indexOf('Feedback:');
-      const score = feedbackResult
-        .substring(score_index + 'Score: '.length, feedback_index)
-        .trim();
-      const feedback = feedbackResult
-        .substring(feedback_index + 'Feedback: '.length)
-        .trim();
-      console.log(score);
-      console.log(feedback);
-
-      const item = {
-        feedbackId: { S: `${Math.random()}` },
-        feedbackScore: { S: score },
-        feedbackText: { S: feedback },
-      };
-      try {
-        const command = new PutItemCommand({
-          TableName: FeedbackTableName,
-          Item: item,
-        });
-        await dynamoClient.send(command);
-        return;
-      } catch (error) {
-        console.log('Error storing result');
-        return;
-      }
+    let status = transcript.TranscriptionJob?.TranscriptionJobStatus;
+    while (status != 'COMPLETED') {
+      await new Promise(resolve => setTimeout(resolve, 1000));
+      const result = await transcribeClient.send(
+        new GetTranscriptionJobCommand({
+          TranscriptionJobName: audioFileName,
+        }),
+      );
+      status = result.TranscriptionJob?.TranscriptionJobStatus;
     }
+  } catch (err) {
+    console.error('Error:', err);
+  }
+}
+
+/*
+  This function retrieves the transcript from the S3 bucket and returns it.
+*/
+async function retrieveTranscript(audioFileName: string) {
+  const input = {
+    Bucket: uploadResponseBucket,
+    Key: `${audioFileName}.json`,
+  };
+  const command = new GetObjectCommand(input);
+  const response = await s3Client.send(command);
+  if (!response.Body) {
+    return {
```

Comment:
Note that using return here will store the status code and body inside variable `answer` in line 39.

One way to solve this in `main` is to check if the function returned string, in that case it would be successful and execution will continue, otherwise that would me there is a failure and we return the object.

Reply:
Fixed in [2b421af](https://github.com/bahrain-uob/gen-ai-moe-challenge/pull/83/commits/2b421af4608af1babeb7864efb8d5710ee051f97)